kvpress/presses/chunk_press.py

```diff
@@ -50,21 +58,80 @@ def compress(
         assert attentions is None, "ChunkPress does not support attentions."
 
         kv_len = keys.shape[2]
-
         indices = []
-        for i in range(0, kv_len, self.chunk_length):
-            chunk_scores = self.press.score(
+
+        if self.global_scoring:
+            # 1. Calculate global scores first
+            global_scores = self.press.score(
                 module,
-                hidden_states[:, i : i + self.chunk_length],
-                keys[:, :, i : i + self.chunk_length],
-                values[:, :, i : i + self.chunk_length],
+                hidden_states,
+                keys,
+                values,
                 attentions,
                 kwargs,
             )
-            chunk_length = keys[:, :, i : i + self.chunk_length].shape[2]
-            n_kept = max(1, int(chunk_length * (1 - self.press.compression_ratio)))
-            chunk_indices = i + chunk_scores.topk(n_kept, dim=-1).indices
-            indices.append(chunk_indices)
+            
+            # 2. Calculate actual number of complete chunks and remaining tokens
+            num_complete_chunks = kv_len // self.chunk_length
+            remaining_tokens = kv_len % self.chunk_length
+            
+            # Reshape complete chunks for score calculation
+            if num_complete_chunks > 0:
```

Comment:
Let's reformulate this:
```
if num_complete_chunks == 0:
    return self.press.compress(...
```
It would also make sense to use a larger threshold, e.g.
`if num_complete_chunks < 10:` to handle short sequences.